tensorboard/plugins/pr_curve/summary_test.py

```diff
@@ -38,56 +39,240 @@ def setUp(self, *args, **kwargs):
+      [0.0, 24.0, 105.0, 144.0, 150.0],  # False negatives.
+      [0.3333333, 0.3853211, 0.5421686, 0.75, 1.0],  # Precision.
+      [1.0, 0.8400000, 0.3000000, 0.0400000, 0.0],  # Recall.
+    ], tensor_events[0])
+    self.validateTensorEvent_(1, [
+      [150.0, 128.0, 45.0, 4.0, 0.0],  # True positives.
+      [300.0, 204.0, 39.0, 6.0, 0.0],  # False positives.
+      [0.0, 96.0, 261.0, 294.0, 300.0],  # True negatives.
+      [0.0, 22.0, 105.0, 146.0, 150.0],  # False negatives.
+      [0.3333333, 0.3855421, 0.5357142, 0.4000000, 1.0],  # Precision.
+      [1.0, 0.8533333, 0.3000000, 0.0266666, 0.0],  # Recall.
+    ], tensor_events[1])
+    self.validateTensorEvent_(2, [
+      [150.0, 120.0, 39.0, 4.0, 0.0],  # True positives.
+      [300.0, 185.0, 38.0, 2.0, 0.0],  # False positives.
+      [0.0, 115.0, 262.0, 298.0, 300.0],  # True negatives.
+      [0.0, 30.0, 111.0, 146.0, 150.0],  # False negatives.
+      [0.3333333, 0.3934426, 0.5064935, 0.6666666, 1.0],  # Precision.
+      [1.0, 0.8000000, 0.2599999, 0.0266666, 0.0],  # Recall.
+    ], tensor_events[2])
+
+  def testWeightedRandomly(self):
+    # Verify that the metadata was correctly written.
+    accumulator = self.multiplexer.GetAccumulator('colors_weighted_randomly')
+    tag_content_dict = accumulator.PluginTagToContent('pr_curves')
 
     # Test the summary contents.
-    tensor_events = accumulator.Tensors('tag_bar/tag_bar')
-    self.assertEqual(1, len(tensor_events))
-    tensor_event = tensor_events[0]
-    self.assertEqual(1, tensor_event.step)
+    expected_tags = ['red/red', 'green/green', 'blue/blue']
+    self.assertItemsEqual(expected_tags, list(tag_content_dict.keys()))
 
-    tensor_nd_array = tf.make_ndarray(tensor_event.tensor_proto)
-    np.testing.assert_allclose([
-      # True positives.
-      [3.0, 3.0, 3.0, 3.0, 2.0, 2.0, 1.0, 1.0, 0.0, 0.0],
-      # False positives.
-      [3.0, 3.0, 2.0, 2.0, 2.0, 2.0, 1.0, 1.0, 0.0, 0.0],
-      # True negatives.
-      [0.0, 0.0, 1.0, 1.0, 1.0, 1.0, 2.0, 2.0, 3.0, 3.0],
-      # False negatives.
-      [0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 2.0, 2.0, 3.0, 3.0],
-      # Precision.
-      [0.5, 0.5, 0.6, 0.6, 0.5, 0.5, 0.5, 0.5, 1.0, 1.0],
-      # Recall.
-      [1.0, 1.0, 1.0, 1.0, 2/3, 2/3, 1/3, 1/3, 0.0, 0.0],
-    ], tensor_nd_array)
+    for tag in expected_tags:
+      # Parse the data within the JSON string and set the proto's fields.
+      plugin_data = pr_curve_pb2.PrCurvePluginData()
+      json_format.Parse(tag_content_dict[tag], plugin_data)
+      self.assertEqual(5, plugin_data.num_thresholds)
+
+      # Test the summary contents.
+      tensor_events = accumulator.Tensors(tag)
+      self.assertEqual(3, len(tensor_events))
+
+    # Test the output for the red classifier. The red classifier has the
+    # narrowest standard deviation.
+    tensor_events = accumulator.Tensors('red/red')
+    self.validateTensorEvent_(0, [
+      [45.7408180, 18.8553009, 5.1260533, 1.3998152, 0.0],  # True positives.
```

Comment:
It unfortunately seems like the PR data for the weighted randomly case differs when I run on my mac despite `tf.set_random_seed(42)`, and I don't quite understand the op level seeds.

I think I'm just going to revert the test from this change, and write a separate PR that makes this test use a deterministic case (instead of relying on the demo). Another advantage of using the previous test example was that we could manually validate whether the PR calculations were correct.